starport/cmd/network_chain_show.go

```diff
@@ -0,0 +1,204 @@
+	case chainShowGenesis:
+		content, err = formatChainGenesis(nb, chainLaunch)
+	case chainShowInfo:
+		content, err = formatChainInfo(cmd.Context(), chainLaunch)
+	case chainShowAccounts:
+		content, err = formatChainAccounts(cmd.Context(), n, launchID)
+	case chainShowPeers:
+		content, err = formatChainPeers(cmd.Context(), n, launchID)
+	}
+	if err != nil {
+		return err
+	}
+
+	nb.Spinner.Stop()
+	fmt.Print(content)
+	return nil
+}
+
+func formatChainGenesis(nb NetworkBuilder, chainLaunch networktypes.ChainLaunch) (string, error) {
+	c, err := nb.Chain(networkchain.SourceLaunch(chainLaunch))
+	if err != nil {
+		return "", err
+	}
+
+	genesisPath, err := c.GenesisPath()
+	if err != nil {
+		return "", err
+	}
+	// TODO run the prepare command in dry run mode and print the genesis if not exist
+	if _, err = os.Stat(genesisPath); os.IsNotExist(err) {
+		return "", errors.New("chain genesis not found, run the 'network chain prepare' command first")
+	}
+	genesisFile, err := os.ReadFile(genesisPath)
+	if err != nil {
+		return "", err
+	}
+	return string(genesisFile), nil
+}
+
+func formatChainInfo(ctx context.Context, chainLaunch networktypes.ChainLaunch) (info string, err error) {
+	var genesis []byte
+	if chainLaunch.GenesisURL != "" {
+		genesis, _, err = cosmosutil.GenesisAndHashFromURL(ctx, chainLaunch.GenesisURL)
+		if err != nil {
+			return "", err
+		}
+	}
+	chainInfo := struct {
+		Chain   networktypes.ChainLaunch `json:"Chain"`
+		Genesis []byte                   `json:"Genesis"`
+	}{
+		Chain:   chainLaunch,
+		Genesis: genesis,
+	}
+	return yaml.Marshal(ctx, chainInfo, "$.Genesis")
+}
+
+func formatChainAccounts(ctx context.Context, n network.Network, launchID uint64) (string, error) {
+	genesisInformation, err := n.GenesisInformation(ctx, launchID)
+	if err != nil {
+		return "", err
+	}
+	accountSummary := bytes.NewBufferString("")
+
+	genesisAccEntries := make([][]string, 0)
+	for _, acc := range genesisInformation.GenesisAccounts {
+		genesisAccEntries = append(genesisAccEntries, []string{
+			acc.Address,
+			acc.Coins,
+		})
+	}
+	if len(genesisAccEntries) > 0 {
+		if err = entrywriter.MustWrite(
+			accountSummary,
+			chainGenesisAccSummaryHeader,
+			genesisAccEntries...,
+		); err != nil {
+			return "", err
+		}
+	}
+
+	genesisVestingAccEntries := make([][]string, 0)
+	for _, acc := range genesisInformation.VestingAccounts {
+		genesisVestingAccEntries = append(genesisVestingAccEntries, []string{
+			acc.Address,
+			acc.StartingBalance,
+			acc.Vesting,
+			strconv.FormatInt(acc.EndTime, 10),
+		})
+	}
+	if len(genesisVestingAccEntries) > 0 {
+		if err = entrywriter.MustWrite(
+			accountSummary,
+			chainVestingAccSummaryHeader,
+			genesisVestingAccEntries...,
+		); err != nil {
+			return "", err
+		}
+	}
+	return accountSummary.String(), err
+}
+
+func formatChainPeers(ctx context.Context, n network.Network, launchID uint64) (string, error) {
+	genesisInformation, err := n.GenesisInformation(ctx, launchID)
```

Comment:
Genesis information will query everything for launch (genesis accounts, vesting accounts and validators ) while only validators are need here. We can use the genesis validators query